features/step_definitions/collection.js

```diff
@@ -0,0 +1,267 @@
+                }
+            });
+    });
+
+    Then(/^the collection is empty$/, function step() {
+        return request({
+            uri: `${this.instance}/${this.namespace}/collections/${this.firstCollectionId}`,
+            method: 'GET',
+            auth: {
+                bearer: this.jwt
+            },
+            json: true
+        }).then((response) => {
+            Assert.strictEqual(response.statusCode, 200);
+            Assert.deepEqual(response.body.pipelineIds, []);
+        });
+    });
+
+    When(/^they update the collection "myCollection" with that pipeline$/, function step() {
+        return this.ensurePipelineExists({ repoName: this.repoName })
+            .then(() => {
+                const pipelineId = parseInt(this.pipelineId, 10);
+
+                request({
+                    uri: `${this.instance}/${this.namespace}/collections/${this.firstCollectionId}`,
+                    method: 'PUT',
+                    auth: {
+                        bearer: this.jwt
+                    },
+                    body: {
+                        pipelineIds: [pipelineId]
+                    },
+                    json: true
+                }).then((response) => {
+                    Assert.strictEqual(response.statusCode, 200);
+                });
+            });
+    });
+
+    Given(/^they have a collection "myCollection"$/, function step() {
+        return createCollection.call(this, { name: 'myCollection' })
+            .then((response) => {
+                Assert.oneOf(response.statusCode, [409, 201]);
+
+                if (response.statusCode === 201) {
+                    this.firstCollectionId = response.body.id;
+                } else {
+                    const str = response.body.message;
+
+                    this.firstCollectionId = str.split(': ')[1];
+                }
+            });
+    });
+
+    Given(/^they have a collection "anotherCollection"$/, function step() {
+        return createCollection.call(this, { name: 'anotherCollection' })
+            .then((response) => {
+                Assert.oneOf(response.statusCode, [409, 201]);
+
+                if (response.statusCode === 201) {
+                    this.firstCollectionId = response.body.id;
+                } else {
+                    const str = response.body.message;
+
+                    this.secondCollectionId = str.split(': ')[1];
+                }
+            });
+    });
+
+    When(/^they fetch all their collections$/, function step() {
+        return request({
+            uri: `${this.instance}/${this.namespace}/collections`,
+            method: 'GET',
+            auth: {
+                bearer: this.jwt
+            },
+            json: true
+        }).then((response) => {
+            Assert.strictEqual(response.statusCode, 200);
+        });
+    });
+
+    Then(/^they can see those collections$/, function step() {
+        return request({
```

Comment:
This should be receiving data from the `/^they fetch all their collections$/` step and not making a separate call.

Reply:
How do I send data from that step to this one?